db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/pre_deployment.rb

```diff
@@ -1,93 +1,70 @@
-require 'net/ssh'
-
 LOCAL_BOOK = 'local_book.yaml'.freeze
-REPO_URL = "https://copr.fedorainfracloud.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo".freeze
+REPO_URL   = "https://copr.fedorainfracloud.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo".freeze
 
 def handle_rhel_subscriptions(commands)
-  system "scp -o 'StrictHostKeyChecking no' rhel_subscribe_inventory.yaml " \
-    "#{$evm.root['user']}@#{$evm.root['deployment_master']}:~/"
-  rhel_subscribe_cmd = "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
-                       "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml"
-  commands.push("sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/", rhel_subscribe_cmd)
-end
-
-def ssh_exec!(ssh, command)
-  stdout_data, stderr_data = "", ""
-  exit_code, exit_signal = nil, nil
-
-  ssh.open_channel do |channel|
-    channel.exec("ssh -A -o 'StrictHostKeyChecking no' -t -t #{$evm.root['user']}@#{$evm.root['deployment_master']} " \
-                 + command) do |_, success|
-      raise StandardError, "Command \"#{command}\" was unable to execute" unless success
+  commands.unshift("sudo subscription-manager repos --disable=\"*\"",
+                   "sudo subscription-manager repos --enable=\"rhel-7-server-rh-common-rpms\" --enable=\"rhel-7-server-rpms\" --enable=\"rhel-7-server-extras-rpms\" --enable=\"rhel-7-server-ose-3.2-rpms\"")
+  $evm.root['container_deployment'].perform_scp($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), "rhel_subscribe_inventory.yaml", "rhel_subscribe_inventory.yaml")
 
-      channel.on_data do |_, data|
-        stdout_data += data
-      end
-
-      channel.on_extended_data do |_, _, data|
-        stderr_data += data
-      end
-
-      channel.on_request("exit-status") do |_, data|
-        exit_code = data.read_long
-      end
-
-      channel.on_request("exit-signal") do |_, data|
-        exit_signal = data.read_long
-      end
-    end
-  end
-  ssh.loop
-  {
-    :stdout      => stdout_data,
-    :stderr      => stderr_data,
-    :exit_code   => exit_code,
-    :exit_signal => exit_signal
-  }
+  commands.push("sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/") if $evm.vmdb(:container_deployment).find($evm.root['automation_task'].automation_request.options[:attrs][:deployment_id]). container_deployment_nodes.count > 1
+  commands
```

Comment:
for side effect methods, it's ruby way to put ! in the end of the method

but yeah this will be better without side effects. Just return new array from each method and collect/call them later like

run_commands(pre_deployment + subscription + package_install ...)

and best to nicely log each group. Given you want to do this all in one ssh connectipn, using above way, the order of commands should be nicely visible.